tools/define-compose-url.sh

```diff
@@ -2,19 +2,20 @@
 set -euo pipefail
 source /etc/os-release
 
-# This isn't needed when not running on RHEL
-if [[ $ID != rhel ]]; then
-  exit 0
+# This isn't needed when not running on RHEL or it's subscribed
+if [[ $ID != rhel ]] || sudo subscription-manager status; then
+  return 0
 fi
 
 if [[ $ID == rhel && ${VERSION_ID%.*} == 8 ]]; then
-  COMPOSE_ID=$(curl -L http://download.devel.redhat.com/rhel-8/nightly/RHEL-8/latest-finished-RHEL-8.6/COMPOSE_ID)
+  COMPOSE_ID=$(curl -L http://download.devel.redhat.com/rhel-8/nightly/RHEL-8/latest-finished-RHEL-"${VERSION_ID}"/COMPOSE_ID)
 
   # default to a nightly tree but respect values passed from ENV so we can test rel-eng composes as well
   COMPOSE_URL="${COMPOSE_URL:-http://download.devel.redhat.com/rhel-8/nightly/RHEL-8/$COMPOSE_ID}"
 
 elif [[ $ID == rhel && ${VERSION_ID%.*} == 9 ]]; then
-  COMPOSE_ID=$(curl -L http://download.devel.redhat.com/rhel-9/nightly/RHEL-9-Beta/latest-RHEL-9.0/COMPOSE_ID)
+  # NOTE: remove Beta from the url once Beta is out
+  COMPOSE_ID=$(curl -L http://download.devel.redhat.com/rhel-9/nightly/RHEL-9-Beta/latest-RHEL-"${VERSION_ID}"/COMPOSE_ID)
```

Comment:
I think Beta is already out, no ?

Reply:
Right, this is from before beta was not yet out, I'll update it, thanks!

Reply:
Updated, we just don't have a `latest-finished` symlink for RHEL-9 yet though.

Reply:
URL looks good but the comment above can be removed b/c it is no longer valid.